baseapp-wagtail/baseapp_wagtail/base/rest_framework/pages/views.py

```
@@ -0,0 +1,62 @@
+        "url_path",
+        "ancestors",
+    ]
+
+    def path_view(self, request):
+        queryset = self.get_queryset()
+        try:
+            obj = self.find_object(queryset, request)
+            if obj is None:
+                raise self.model.DoesNotExist
+        except self.model.DoesNotExist:
+            raise Http404("not found")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)